### PR TITLE
Bugfix: create view from another view

### LINKED ISSUE #636: pointer to a view to an array is allocatable
**Describe the bug**
the attribute allocatable is True when it should be false
when assigning a view to another variable

**To Reproduce**
```python
import numpy as np
a = np.array([[1, 2, 3, 4, 5], [6, 7, 8, 9, 10]])
c = a[1:]
b = c[1:]
b = a
```

Provide the generated code, or the error message:
```bash/fortran/c
Arrays which own their data cannot become views on other arrays (b)
```

**Expected behavior**
we should not raise an error here as b is not an allocatable

### PR DESCRIPTION
**Pull request by saniQQ**
Fix #636:

- Fix bug where view created from another view has wrong values of attributes `is_allocatable` & `is_pointer`;
- Fix bug where `is_target` attribute is set to true when target is pointer.

### CODE REVIEW

**Review by mostafaMamoni — APPROVED** [review]

File: pyccel/parser/semantic.py [hunk 1]
```diff
@@ -1364,13 +1364,13 @@ def _ensure_target(self, rhs, d_lhs):
...
             rhs.is_target = True
-        if isinstance(rhs, IndexedElement) and rhs.rank > 0 and rhs.base.allocatable:
+        if isinstance(rhs, IndexedElement) and (rhs.rank > 0 and (rhs.base.allocatable or rhs.base.is_pointer)):
             d_lhs['allocatable'] = False
             d_lhs['is_pointer' ] = True
 
             # TODO uncomment this line, to make rhs target for
             #      lists/tuples.
-            rhs.base.is_target = True
+            rhs.base.is_target = True if rhs.base.allocatable else False
```
Comment (EmilyBourne) [inline pyccel/parser/semantic.py:1373]: What about stack arrays?

Comment (saniQQ) [reply pyccel/parser/semantic.py:1373]: i think i will change it to be True if rhs.base.is_pointer is false

File: pyccel/parser/semantic.py [hunk 2]
```diff
@@ -1364,13 +1364,13 @@ def _ensure_target(self, rhs, d_lhs):
             # TODO uncomment this line, to make rhs target for
             #      lists/tuples.
             rhs.is_target = True
-        if isinstance(rhs, IndexedElement) and rhs.rank > 0 and rhs.base.allocatable:
+        if isinstance(rhs, IndexedElement) and (rhs.rank > 0 and (rhs.base.allocatable or rhs.base.is_pointer)):
```
Comment (MenoIy) [inline pyccel/parser/semantic.py:1367]: extra parentheses

**Review by MenoIy — CHANGES_REQUESTED** [review]
can you add some tests ?

**Review by MenoIy — APPROVED** [review]

**Review by EmilyBourne — APPROVED** [review]

**Review by yguclu — APPROVED** [review]

**Pull request merged**